# Task 2

## Answer 2.a

Using the definition of VaR sensitivity,

$$\frac{\partial VaR(\omega)}{\partial \omega}=\mu_i+z_\alpha * \frac{(\sum\omega)_i}{\sqrt{\omega'\sum\omega}}$$

the result can be verified by computing the **sensitivity vector** and comparing it with the values stated in the question.

$$ sensitivity\_vector = \mu + z_\alpha * \frac{\sum\omega}{\sqrt{\omega'\sum\omega}}$$

The calculation therefore begins with $\sum\omega$ and $\sqrt{\omega'\sum\omega}$.

The quantities provided in the question are defined below.

$\mu$ vector: 

$$
mu=\begin{pmatrix}
0 \\
0 \\
0 \\
\end{pmatrix}
$$

$\sigma$ vector:

$$
sd=\begin{pmatrix}
0.30 \\
0.20 \\
0.15 \\
\end{pmatrix}
$$

$$
S=\begin{pmatrix}
0.30 & 0 & 0\\
0 & 0.20 & 0 \\
0 & 0 & 0.15 \\
\end{pmatrix}
$$

$\omega$ vector:

$$
w=\begin{pmatrix}
0.50 \\
0.20 \\
0.30 \\
\end{pmatrix}
$$

$\rho$ matrix:

$$
R=\begin{pmatrix}
1.0 & 0.8 & 0.5 \\
0.8 & 1.0 & 0.3 \\
0.5 & 0.3 & 1.0 \\
\end{pmatrix}
$$

In [67]:
import numpy as np

mu = np.array([[0],
               [0],
               [0]])

sd = np.array([[0.30],
               [0.20],
               [0.15]])

S = np.diag(sd.flatten())

w = np.array([[0.5],
              [0.2],
              [0.3]])

R = np.array([[1.0, 0.8, 0.5],
              [0.8, 1.0, 0.3],
              [0.5, 0.3, 1.0]])

This gives
$$\sum=SRS'=SRS=
\begin{pmatrix}
0.0900	& 0.0480	& 0.0225 \\
0.0480	& 0.0400	& 0.0090 \\
0.0225	& 0.0090	& 0.0225 \\
\end{pmatrix}
$$

In [68]:
SRS = S @ R @ S
SRS

array([[0.09  , 0.048 , 0.0225],
       [0.048 , 0.04  , 0.009 ],
       [0.0225, 0.009 , 0.0225]])

Hence,

$$\sum\omega=
\begin{pmatrix}
0.0900	& 0.0480	& 0.0225 \\
0.0480	& 0.0400	& 0.0090 \\
0.0225	& 0.0090	& 0.0225 \\
\end{pmatrix}
\begin{pmatrix}
0.5 \\
0.2 \\
0.3 \\
\end{pmatrix}
=\begin{pmatrix}
0.0614 \\
0.0347 \\
0.0198 \\
\end{pmatrix}
$$

In [69]:
sigma_w = SRS @ w
sigma_w

array([[0.06135],
       [0.0347 ],
       [0.0198 ]])

Moreover,

$$
\omega'\sum\omega=
\begin{pmatrix}
0.5 \\
0.2 \\
0.3 \\
\end{pmatrix}'
\begin{pmatrix}
0.0900	& 0.0480	& 0.0225 \\
0.0480	& 0.0400	& 0.0090 \\
0.0225	& 0.0090	& 0.0225 \\
\end{pmatrix}
\begin{pmatrix}
0.5 \\
0.2 \\
0.3 \\
\end{pmatrix}
=0.043555
$$

$$\sqrt{\omega'\sum\omega}=0.20869835$$

In [72]:
w_sigma_w = w.T @ sigma_w
w_sigma_w_sqrt = np.sqrt(w_sigma_w)
print(f'w_sigma_w: {w_sigma_w}, w_sigma_w_sqrt: {w_sigma_w_sqrt}')

w_sigma_w: [[0.043555]], w_sigma_w_sqrt: [[0.20869835]]


Finally, we calculate $sensitivity\_vector = \mu + z_\alpha * \frac{\sum\omega}{\sqrt{\omega'\sum\omega}}$.

In [76]:
z = -2.326
sensitivity = mu + z * sigma_w / w_sigma_w_sqrt
sensitivity

array([[-0.68376248],
       [-0.38674096],
       [-0.2206764 ]])

Now we obtain the three sensitivities:

$$ sensitivity_1 = -0.68376248$$
$$ sensitivity_2 = -0.38674096$$
$$ sensitivity_3 = -0.2206764 $$

These values agree with the sensitivities given in the question.

---


## Answer 2.b
1. All three assets are positively correlated, so increasing the weight of any asset raises overall portfolio volatility and tail risk.
In the formula, $z_\alpha$ is negative, and the magnitude of the risk term dominates the expected return term, making marginal VaR and marginal ES negative.

2. Negative sensitivities mean that increasing the weight of any asset causes the portfolio VaR to become more negative, representing a larger potential loss and higher portfolio risk.


---


## Answer 2.c
Asset 1 has both the highest volatility and the largest portfolio weight, which gives it the **largest** marginal impact on portfolio risk. Therefore, it exhibits the most negative sensitivity.

---

## Answer 2.d
#### Mathematical Explanation:

Since $\mu_i = 0$, we only need to compare the absolute values of $\lvert z_\alpha \rvert$ and $\left|\frac{\phi(z_\alpha)}{1-c}\right|$.

Since

$$\lvert z_\alpha \rvert=2.326$$

$$\left|\frac{\phi(z_\alpha)}{1-c}\right|=\left|\frac{0.02667}{0.01}\right|=2.667$$

Thus

$$\left|\frac{\phi(z_\alpha)}{1-c}\right| > \lvert z_\alpha \rvert$$

Therefore, $\lvert \frac{\partial ES}{\partial \omega_i}\rvert > \lvert \frac{\partial VaR}{\partial \omega_i}\rvert$


#### Economic Interpretation:
VaR only considers the quantile level, while ES considers the average loss in the tail and is therefore more sensitive to tail risk.

Therefore, changes in marginal weights have a larger impact on ES.